{{ cookiecutter.repo_name }}/{{ cookiecutter.repo_name }}/settings.py

```diff
@@ -18,8 +18,6 @@
 
 load_dotenv()
 
-BASE = os.path.dirname(__file__)
-
 class Settings(TypedDict):
     DATA_PATH: Path
```

Comment:
I am not fan of TypeDict can we use dataclass-es?

Reply:
It was proposed by @pdeszynski

Reply:
Both will have their advantage and disadvantage. For e.g. using dataclasses only it will be cumbersome to do updates (for e.g. changing only 1 setting from a dataclass during initialization.

There will be a more scaffolding with dataclasses, but at the end I was using both to have simple API for updates, and dataclass to store settings.



```python
BASE = Path(__file__).parent.parent


class UpdateSettings(TypedDict, total=False):
    DATA_PATH: Path

@dataclass
class Settings:
    DATA_PATH: Path
    OTHER_PATH: Path = field(init=False)

    def __post_init__(self):
        self.OTHER_PATH = Path(os.getenv("OTHER_PATH", self.PATH / "other_path"))

    def update(self, new_settings: UpdateSettings):
        for key, value in new_settings.items():
            setattr(self, key, value)

        self.__post_init__()


settings = Settings(
    DATA_PATH=BASE / "data_path",
)


def update_settings(new_settings: UpdateSettings):
    settings.update(new_settings)
```